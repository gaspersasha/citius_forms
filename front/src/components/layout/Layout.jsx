import React from 'react';
import PropTypes from 'prop-types';
import { useRouter } from 'next/router';
import cn from 'classnames';
import { useAmp } from 'next/amp';
import {
  Header,
  Footer,
  AmpHeader,
  AmpSideBar,
  LoginGateway,
  Breadcrumbs,
} from '~components';
import { DeviceContextProvider } from '~contexts';
import { GlobalBody } from '~services';
import s from './layout.module.sass';

// Paths where header and footer doesn't render
export const emptyPaths = [
  '/goreserve',
  '/reservation',
  '/gofinance',
  '/terms',
  '/soft-check',
  '/finance-eligibility',
  '/almost-done',
  '/part-ex',
  '/contact-dealer',
];

export const clickablePaths = ['/finance-eligibility'];

export const breadcrumbsPaths = [
  '/account',
  '/finance-thank-you',
  // temp. just for these paths due to this ticket:
  // https://creativesolutions.atlassian.net/browse/BE-763
];

export const config = {
  amp: 'hybrid',
};

const Layout = ({ children }) => {
  const isAmp = useAmp();
  const { pathname } = useRouter();
  const isEmptyPath = emptyPaths.includes(pathname);
  const clickableLogo = clickablePaths.includes(pathname);
  const withBreadcrumbs = breadcrumbsPaths.includes(
    `/${pathname.split('/')[1]}`
  );

  // general head should be put here too

  const content = pathname.includes('/account') ? (
    <LoginGateway>{children}</LoginGateway>
  ) : (
    children
  );

  return (
    <>
      {isAmp ? (
        <>
          <AmpSideBar />
          <AmpHeader isEmpty={isEmptyPath} />
          {content}
        </>
      ) : (
        <>
          <DeviceContextProvider>
            <Header isEmpty={isEmptyPath} clickableLogo={clickableLogo} />
          </DeviceContextProvider>
          <div className={cn(s.content, { 'empty-header': isEmptyPath })}>
            {withBreadcrumbs && <Breadcrumbs />}
            {content}
          </div>
          {!isEmptyPath && <Footer />}
          <GlobalBody pageType={pathname} />
        </>
      )}
    </>
  );
};

Layout.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Layout;
