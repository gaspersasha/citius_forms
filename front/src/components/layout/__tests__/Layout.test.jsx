import React from 'react';
import { useRouter } from 'next/router';

import Layout from '../Layout';

jest.mock('next/router', () => ({
  useRouter: jest.fn(),
}));

useRouter
  .mockImplementation(() => ({
    pathname: '/',
  }))
  .mockImplementationOnce(() => ({
    pathname: '/goreserve',
  }))
  .mockImplementationOnce(() => ({
    pathname: '/reservation',
  }))
  .mockImplementationOnce(() => ({
    pathname: '/gofinance',
  }));

describe('<Layout />', () => {
  let wrapper;

  const getWrapper = () =>
    shallow(
      <Layout>
        <div>Test</div>
      </Layout>
    );

  beforeEach(() => {
    wrapper = getWrapper();
  });

  it('header and footer doesn`t rendered on /goreserve route', () => {
    expect(wrapper.find('[data-id="menu"]').exists()).toBeFalsy();
    expect(wrapper.find('footer').exists()).toBeFalsy();
  });

  it('header and footer doesn`t rendered on /reservation route', () => {
    expect(wrapper.find('[data-id="menu"]').exists()).toBeFalsy();
    expect(wrapper.find('footer').exists()).toBeFalsy();
  });

  it('header and footer doesn`t rendered on /gofinance route', () => {
    expect(wrapper.find('[data-id="menu"]').exists()).toBeFalsy();
    expect(wrapper.find('footer').exists()).toBeFalsy();
  });

  it('rendered without errors', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
