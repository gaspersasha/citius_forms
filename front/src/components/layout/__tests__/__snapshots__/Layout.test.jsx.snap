// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> rendered without errors 1`] = `
<Fragment>
  <DeviceContextProvider>
    <Header
      clickableLogo={false}
      isEmpty={false}
    />
  </DeviceContextProvider>
  <div
    className="content"
  >
    <div>
      Test
    </div>
  </div>
  <Footer />
  <GlobalBody
    pageType="/"
  />
</Fragment>
`;
